proxy/InkAPI.cc

```diff
@@ -1053,15 +1053,14 @@ int
 INKVConnInternal::handle_event(int event, void *edata)
 {
   handle_event_count(event);
-  if (m_deleted) {
-    if (m_deletable) {
-      this->mutex = NULL;
-      m_read_vio.set_continuation(NULL);
-      m_write_vio.set_continuation(NULL);
-      INKVConnAllocator.free(this);
-    }
-  } else {
+  // If the VConn isn't deleted, call the handler
+  if (!m_deleted) {
     return m_event_func((TSCont) this, (TSEvent) event, edata);
+  } else {
```

Comment:
@igalic This handler shouldn't be called after the VConn was destroyed. The else exists soely to get the assert in there (for debug builds)-- because if we hit that `else` path there is a bug-- but we don't want ATS to crash on it if possible.